### PR TITLE
Fix variable names in BatchNorm and LayerNorm implementations

### PR DESCRIPTION
**Pull request by yan12125**
**Description**: Describe your changes.

* Use actual variable names in comments
* Make parentheses balanced

**Motivation and Context**

I was studying how onnxruntime implements BatchNorm, and got confused by comments. I found similar issues in comments for LayerNorm, so I fixed them all in this PR. Hopefully it makes the code clearer.

### CODE REVIEW

**Comment by SherlockNoMad** [comment]
Could you do a global search of "inv_std_var"? I believe most of them should "inv_std"

File: orttraining/orttraining/test/training_ops/cuda/layer_norm_test.cc [hunk 1]
```diff
@@ -173,7 +173,7 @@ static void TestInvertibleLayerNormGrad(
       inv_std_var_data[i] = X.col(i).square().mean() - mean_data[i] * mean_data[i];
     }
 
-    // Compute Y = ((x - mean) * (inv_var) * scale + bias
+    // Compute Y = ((x - mean) * (inv_std_var)) * scale + bias
```
[the remark this answers is not in the document]
Comment (yan12125) [reply orttraining/orttraining/test/training_ops/cuda/layer_norm_test.cc:176]: Thanks for the hint. Indeed this should be `inv_std`. I'd like to run tests with CUDA to make sure norhing is broken after changing variable names, but I got some compile issues. I will try a few more days on running tests.

**Comment by yan12125** [comment]
> Could you do a global search of "inv_std_var"? I believe most of them should "inv_std"

I checked all `inv_std_var` and all of them are indeed standard deviations instead of variances. I've also fixed related variable names/comments/documentation in touched files. Some notes:

* In `onnxruntime/core/graph/contrib_ops/contrib_defs.cc`, some variables are called `InvStdDev`. Is it better to also replace them with `inv_std`?
* Seems neither the CPU [1] nor the CUDA [2] implementation of `SkipLayerNormalization` fills the third output (`inv_std`). I still changed to standard deviation here to be consistent with other normalization operators.
* I didn't touch files under docs/ as those files are automatically generated

[1] `onnxruntime/contrib_ops/cpu/skip_layer_norm.cc`
[2] `onnxruntime/contrib_ops/cuda/bert/skip_layer_norm.cc`

File: orttraining/orttraining/test/training_ops/cuda/layer_norm_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (yan12125) [reply orttraining/orttraining/test/training_ops/cuda/layer_norm_test.cc]: OK I managed to build onnxruntime with CUDA enabled. I've also updated this occurrence (see the comment below).

**Comment by yan12125** [comment]
Rebased to fix conflicts with https://github.com/microsoft/onnxruntime/pull/7626

File: onnxruntime/contrib_ops/cuda/layer_norm.cc [hunk 2]
```diff
@@ -72,32 +72,32 @@ Status LayerNorm<T, U, simplified>::ComputeInternal(OpKernelContext* ctx) const
...
     if (mean != nullptr) {
       mean_data = reinterpret_cast<CudaU*>(mean->template MutableData<U>());
     }
   }
 
-  Tensor* var = ctx->Output(output_index, TensorShape(mean_inv_std_var_dim));
-  CudaU* inv_var_data = nullptr;
-  if (var != nullptr) {
-    inv_var_data = reinterpret_cast<CudaU*>(var->template MutableData<U>());
+  Tensor* std = ctx->Output(output_index, TensorShape(mean_inv_std_dim));
```
Comment (mindest) [inline onnxruntime/contrib_ops/cuda/layer_norm.cc:94]: I think this should also be changed into `inv_std`.

**Comment by mindest** [comment]
@yan12125 Thanks for the contribution and careful work! Please also help check the following files:
- `onnxruntime/core/providers/cuda/nn/batch_norm.cc`: `saved_inv_var_data` to `saved_inv_std_data`, `saved_var` to `saved_inv_std`.
- `orttraining/orttraining/training_ops/cuda/nn/layer_norm_impl.cu`: `c_invvar` to `c_inv_std` (lines 344&386).

> I didn't touch files under docs/ as those files are automatically generated

It may be needed to run [gen_contrib_doc.py](https://github.com/microsoft/onnxruntime/blob/master/tools/python/gen_contrib_doc.py) and [gen_opkernel_doc.py](https://github.com/microsoft/onnxruntime/blob/master/tools/python/gen_opkernel_doc.py) to regenerate the `ContribOperators.md` and `OperatorKernels.md`, then add these updated docs too.

> In `onnxruntime/core/graph/contrib_ops/contrib_defs.cc`, some variables are called `InvStdDev`. Is it better to also replace them with `inv_std`?

Not sure about this.. We intended to use abbreviation `std` for "standard deviation", but `StdDev` is also reasonable. Actually there are also many `std_dev`s present in the codes. @SherlockNoMad Should this be changed too to keep consistent?

**Comment by yan12125** [comment]
Thanks! I addressed most of review comments. However,

> It may be needed to run gen_contrib_doc.py and gen_opkernel_doc.py

Somehow I got errors 

```
$ python tools/python/gen_opkernel_doc.py --output_path output
Traceback (most recent call last):
  File "/home/yen/var/Projects/onnxruntime/tools/python/gen_opkernel_doc.py", line 173, in <module>
    main(args.output_path, args.providers)
  File "/home/yen/var/Projects/onnxruntime/tools/python/gen_opkernel_doc.py", line 73, in main
    opdef = rtpy.get_all_operator_schema()
AttributeError: module 'onnxruntime.capi.onnxruntime_pybind11_state' has no attribute 'get_all_operator_schema'

$ python tools/python/gen_contrib_doc.py                      
Traceback (most recent call last):
  File "/home/yen/var/Projects/onnxruntime/tools/python/gen_contrib_doc.py", line 20, in <module>
    from onnxruntime.capi.onnxruntime_pybind11_state import schemadef  # noqa: F401
ImportError: cannot import name 'schemadef' from 'onnxruntime.capi.onnxruntime_pybind11_state' (/usr/lib/python3.9/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so)
```
I include `-Donnxruntime_PYBIND_EXPORT_OPSCHEMA=ON` in cmake options, and I can find strings `schemadef` and `get_all_operator_schema` in /usr/lib/python3.9/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so. Not sure if I'm doing it right.

**Comment by mindest** [comment]
@yan12125 Hmm, seems that the docs are not required to be updated every time new changes are introduced. If necessary later, try the build command:

```bash
cd onnxruntime
bash build.sh --config Release --parallel --build_wheel --skip_tests --gen_doc
```
which will regenerate the docs when building onnxruntime (can also try the above `python *.py` for regenerating after the build). 

BTW a CI test is failing, could you please try pulling master first?

**Comment by yan12125** [comment]
Thanks for the tips about docs! Will try it out in a few days.

> BTW a CI test is failing, could you please try pulling master first?

Sure, I rebased on top of the current master (https://github.com/microsoft/onnxruntime/commit/a1021a1cf4bca68a975b2ebf4aed57f1b9a3fe4f).

**Comment by yan12125** [comment]
From https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=495411&view=logs&jobId=7e93ae65-9061-5fc2-a6a0-231bb749ebf5&j=7e93ae65-9061-5fc2-a6a0-231bb749ebf5&t=c0b4af22-4c8c-5fb4-565e-430188cb5fa0
```
b"diff --git a/docs/ContribOperators.md b/docs/ContribOperators.md\nindex 44d6c8ae4..5b6077feb 100644\n--- a/docs/ContribOperators.md\n+++ b/docs/ContribOperators.md\n@@ -2678,8 +2678,8 @@ This version of the operator has been available since version 1 of the 'com.micr\n <dd>3D output tensor with shape (batch_size, sequence_length, hidden_size)</dd>\n <dt><tt>mean</tt> (optional) : U</dt>\n <dd>Saved mean used during training to speed up gradient computation</dd>\n-<dt><tt>inv_std_var</tt> (optional) : U</dt>\n-<dd>Saved inverse standard variance used during training to speed up gradient computation.</dd>\n+<dt><tt>inv_std</tt> (optional) : U</dt>\n+<dd>Saved inverse standard deviation used during training to speed up gradient computation.</dd>\n </dl>\n \n #### Type Constraints\n@@ -2688,7 +2688,7 @@ This version of the operator has been available since version 1 of the 'com.micr\n <dt><tt>T</tt> : tensor(float), tensor(float16)</dt>\n <dd>Constrain input and output types to float or half tensors.</dd>\n <dt><tt>U</tt> : tensor(float)</dt>\n-<dd>Constrain mean and inv_std_var to float tensors.</dd>\n+<dd>Constrain mean and inv_std to float tensors.</dd>\n </dl>\n \n \n"
2021-09-14 00:57:19,538 build [ERROR] - Generated documents have diffs. Check build output for details.
```
Seems I still need to update docs to make CI happy :)

**Comment by mindest** [comment]
> Seems I still need to update docs to make CI happy :)

@yan12125 Seems so.. please help add that change too :)

**Comment by yan12125** [comment]
Sure, but I don't have time to rebuild onnxruntime until weekends. Will update this as soon as possible :+1:

**Comment by yan12125** [comment]
Finally got it working! This is the final command I used:
```
LD_LIBRARY_PATH=/usr/local/cuda-10.0/lib64 SYSTEM_COLLECTIONURI=https://example.com python3.6 "$PWD/tools/ci_build/build.py" --build_dir "$PWD/build/Linux" --config Release --parallel 2 --build_wheel --skip_tests --gen_doc --use_cuda --cuda_home=/usr/local/cuda-10.0 --cudnn_home=/usr/local/cuda-10.0
```
Some notes:
* I need to enable CUDA, or the section `CUDAExecutionProvider` in docs/OperatorKernels.md is gone
* `SYSTEM_COLLECTIONURI=https://example.com` is added to prevent `-Werror`, which breaks building some CUDA sources

The updated OperatorKernels.md misses `bfloat16` from type constraints. Not sure why :confused:

**Comment by tianleiwu** [comment]
> LD_LIBRARY_PATH=/usr/local/cuda-10.0/lib64 SYSTEM_COLLECTIONURI=https://example.com python3.6 
> The updated OperatorKernels.md misses `bfloat16` from type constraints. Not sure why 😕

bfloat16 need cuda 11

**Comment by yan12125** [comment]
> bfloat16 need cuda 11

Thanks very much! Now everything looks good!

**Comment by yan12125** [comment]
I'm not sure why I got failures for the ROCm build https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=499012&view=results. Maybe some fixes from git-master are needed - rebased to retry.

**Comment by yan12125** [comment]
Thanks for merging with master! I happened to also look at the merge conflicts today. It's caused by https://github.com/microsoft/onnxruntime/pull/9160, and it did not touch classes in `docs/ContribOperators.md` and `docs/OperatorKernels.md`.

**Comment by yan12125** [comment]
Rebased to fix conflicts after https://github.com/microsoft/onnxruntime/pull/9889

**Pull request closed without merging**